pkg/scalers/http_scaler.go

```diff
@@ -0,0 +1,145 @@
+package scalers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"k8s.io/api/autoscaling/v2beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+	"net/http"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"strconv"
+	"strings"
+)
+
+const (
+	httpMetricName = "HTTPMetricValue"
+)
+
+type httpScaler struct {
+	metadata *httpScalerMetadata
+}
+
+type httpScalerMetadata struct {
+	targetValue int
+	apiURL      string
+	metricName  string
+}
+
+type metric struct {
+	Name  string  `json:"name"`
+	Value float64 `json:"value"`
+}
+
+var httpLog = logf.Log.WithName("http_scaler")
+
+// NewHTTPScaler creates a new HTTP scaler
+func NewHTTPScaler(resolvedEnv, metadata, authParams map[string]string) (Scaler, error) {
+	meta, err := parseHTTPMetadata(resolvedEnv, metadata, authParams)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing HTTP metadata: %s", err)
+	}
+	return &httpScaler{metadata: meta}, nil
+}
+
+func parseHTTPMetadata(resolvedEnv, metadata, authParams map[string]string) (*httpScalerMetadata, error) {
+	meta := httpScalerMetadata{}
+
+	if val, ok := metadata["targetValue"]; ok {
+		targetValue, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("targetValue parsing error %s", err.Error())
+		}
+		meta.targetValue = targetValue
+	} else {
+		return nil, fmt.Errorf("no targetValue given in metadata")
+	}
+
+	if val, ok := metadata["apiURL"]; ok {
+		// remove ending / for better string formatting
+		meta.apiURL = strings.TrimSuffix(val, "/")
+	} else {
+		return nil, fmt.Errorf("no apiURL given in metadata")
+	}
+
+	if val, ok := metadata["metricName"]; ok {
+		meta.metricName = val
+	} else {
+		return nil, fmt.Errorf("no metricName given in metadata")
+	}
+
+	return &meta, nil
+}
+
+func (s *httpScaler) checkHealth() error {
+	u := fmt.Sprintf("%s/health/", s.metadata.apiURL)
+	_, err := http.Get(u)
+	return err
+}
+
+func (s *httpScaler) getMetricInfo() (*metric, error) {
+	var m *metric
+	u := fmt.Sprintf("%s/metrics/%s/", s.metadata.apiURL, s.metadata.metricName)
+	r, err := http.Get(u)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Body.Close()
+	b, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(b, &m)
+	if err != nil {
+		return nil, err
+	}
+	return m, nil
+}
+
+// Close does nothing in case of httpScaler
+func (s *httpScaler) Close() error {
+	return nil
+}
+
+// IsActive returns true if there are pending messages to be processed
+func (s *httpScaler) IsActive(ctx context.Context) (bool, error) {
```

Comment:
I don't think we should fire error everytime when the scaler is not Active.

How is it handled situation when the enpoint return 0 as a value?

Reply:
I think that we should fire an error because that means the API is down. In my opinion that should happens rarely if we use this API to autoscale external services.

> How is it handled the situation when the endpoint return 0 as a value?

Should we do `max(value, 1)` to avoid edge cases? Eventually add multiplier it the API values are [0, 1] range?

Reply:
The `IsActive()` function should return `true` if there are some metrics that we could use for scaling (up), if there aren't any metrics (for example in Kafka scaler, there aren't any messages in the topic) the function should return `false`.

Currently your funcition return `false` only if the enpoint is not available. So how would scaling to 0 look like in your scenario? (only if the enpoint is not available?)

Reply:
Oh, got it! We should check if the metric value is > 0 then. I can see some edge cases in case of continuous values where `0 ~ 0.01` but imho that's something we can "fix" once get some users feedback. Definitely I will cover this in docs